service-workers/service-worker/iframe-controller.https.html

```diff
@@ -0,0 +1,125 @@
+<!DOCTYPE html>
+<title>Service worker inheritance</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+
+let active_workers;
+let parent_controller;
+
+promise_test(t => {
+    const script = 'iframe-controller-worker.https.js';
+    const scope1 = 'iframe-controller.https.html';
+    const scope2 = 'resources/blank.html';
+    const script1 = script + '?for_scope1';
+    const script2 = script + '?for_scope2';
+
+    active_workers = service_worker_unregister_and_register(t, script1, scope1)
+      .then(r => {
+          add_completion_callback(_ => r.unregister());
+          return wait_for_state(t, r.installing, 'activated');
+        })
+      .then(_ => {
+          return service_worker_unregister_and_register(t, script2, scope2);
+        })
+      .then(r => {
+          add_completion_callback(_ => r.unregister());
+          return wait_for_state(t, r.installing, 'activated');
+        });
+
+    return active_workers.then(_ => {
+        // This client's active service worker is set to scope1's registration's
+        // active worker by clients.claim() during the activation.
+        parent_controller = navigator.serviceWorker.controller;
+        return with_iframe('')
```

Comment:
Maybe adding an "options" parameter to with_iframe with an auto_remove option (like in the [chromium version of with_iframe](https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/http/tests/serviceworker/resources/test-helpers.js?q=test-helpers.js+servicew&sq=package:chromium&l=63)) would make sense? Having a boolean that defaults to true like in the chromium version seems wrong, but having an option to pass to with_iframe to get auto remove behavior does seem reasonable.